account_multi_foreign_vat/models/account_fiscal_position.py

```diff
@@ -0,0 +1,39 @@
+from odoo import _, api, models
+from odoo.exceptions import ValidationError
+
+
+class AccountFiscalPosition(models.Model):
+    _inherit = "account.fiscal.position"
+
+    @api.constrains("country_id", "foreign_vat")
+    def _validate_foreign_vat(self):
+        for record in self:
+            try:
+                super()._validate_foreign_vat()
+            except Exception as e:
+                checked_country_code = self.env["res.partner"]._run_vat_test(
+                    record.foreign_vat, record.country_id
+                )
+
+                if (
+                    checked_country_code
+                    and record.country_id
+                    and checked_country_code != record.country_id.code.lower()
+                ):
+                    raise ValidationError(
+                        _(
+                            "The country detected for this foreign VAT "
+                            "number does not match the one set on this fiscal position."
+                        )
+                    ) from e
+                if not checked_country_code and not record.country_id:
+                    raise ValidationError(
+                        _("The foreign VAT number is not correct.")
+                    ) from e
```

Comment:
Is this the only change done to the standard method? https://github.com/odoo/odoo/blob/7d678cc7f4c9e00d78d4120f703e90bd282c1a51/addons/base_vat/models/account_fiscal_position.py